src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1311,6 +1321,13 @@ static ucs_status_t uct_ib_mlx5dv_md_open(struct ibv_device *ibv_device,
     dev->flags    |= UCT_IB_DEVICE_FLAG_MLX5_PRM;
     md->super.name = UCT_IB_MD_NAME(mlx5);
 
+#ifndef HAVE_DC_DV
```

Comment:
why?

Reply:
@yosefe One dummy QP is created in [uct_ib_mlx5dv_check_dc](https://github.com/openucx/ucx/blob/v1.12.1/src/uct/ib/mlx5/dv/ib_mlx5dv_md.c#L915) if the macro ```HAVE_DC_DV``` is defined and [uct_ib_mlx5dv_check_dc](https://github.com/openucx/ucx/blob/v1.12.1/src/uct/ib/mlx5/dv/ib_mlx5dv_md.c#L1031) is called in  ```uct_ib_mlx5dv_md_open```

Since one dummy QP has been created in ```uct_ib_mlx5dv_check_dc```, so it could use that QP to check wether the device support ECE operation. If ```HAVE_DC_DV``` is not defined, the dummy QP won't be created in ```uct_ib_mlx5dv_check_dc```, so it needs to use the help function ```uct_ib_device_ece_check``` to check whether the device support ECE operation.

Should we always use the help function ```uct_ib_device_ece_check``` to check whether the device support ECE operation even that there's already existed dummy QP could be used?

Reply:
seems counter-intuitive.. IMO better check ECE first, then also check DC+ece.

Reply:
Done. Unify to only use ```uct_ib_device_ece_check``` to check it.